app/models/submission.rb

```diff
@@ -409,6 +409,14 @@ def company_name=(value)
     self.company = Company.where("LOWER(name) = LOWER(?)", value).first_or_initialize(name: value)
   end
 
+  def already_voted(current_user)
+    if current_user
+      current_user.votes.exists?(:submission_id => id) 
```

Comment:
This is a total nit, but we've generally moved to the newer-style colon syntax for hash literals in the rest of the codebase, might make sense to do the same here.

```suggestion
      current_user.votes.exists?(submission_id: id) 
```